src/core/unit.c

```diff
@@ -6375,6 +6379,92 @@ Condition *unit_find_failed_condition(Unit *u) {
         return failed_trigger && !has_succeeded_trigger ? failed_trigger : NULL;
 }
 
+int unit_can_live_mount(Unit *u, const char *src, const char *dst, sd_bus_error *error) {
```

Comment:
hmm, i am a bit puzzled about this function.

We currently have unit_can_start(), unit_can_stop(), unit_can_isolate(), unit_can_freeze(), unit_can_clean(). And they all return "universal" truths. i.e. depending on the loaded configuration they decide whether the suggested operation can be executed *at* *all*. They do *not* take current state into account (except for the load state, i.e. whether configuration was acquired at all).

I think this one should behave similar, i.e. not take unit_active_state() into account.

And I think this function should exposed as dbus property of the unit, just like the other others, i.e. there is CanFreeze and CanStart and so on on all units.

Reply:
or let me say this differently: i think it makes sense to refuse live mounts on units before PK if we know for sure that we could not execute it. But i think the actual state checks should not be done before PK, but just after, as part of unit_live_mount() checks.

Reply:
mmh now I wonder, the loaded configuration _could_ change too, no? Given polkit is async, it's possible that a daemon-reload comes in before we get the answer back? In that case the pre-check should pretty much only see if the unit type supports it?

Reply:
So I'll move unit_can_live_mount before even the selinux check, and make it only check really static stuff, like the unit type. And do all the other checks that can mutate in the actual call.

Reply:
but other can_foo do check configuration items, so those can change too before a polkit call returns, what's the strategy there? I'll make this the same for now

Reply:
ok I've added CanLiveMount and made it only check static config data, the variable stuff is checked in the actual mount method now